### DISCUSSION TITLE
Combining multiple blocking and linking passes in splink

### DISCUSSION THREAD

**Discussion by mv802** [moj-analytical-services/splink#2564] [Q&A]
I would like to know how to effectively combine multiple blocking and linking strategies together in the same splink code in such a way that each pass takes the previous pass's unlinked records on works on them subsequently to eventually link all the possible records.

**Comment by RobinL** [comment]
To do this, you'd have to write a loop around the linker, that called `predict()` multiple times.

In general, we recommend using multiple blocking rules in a single model see [here](https://moj-analytical-services.github.io/splink/demos/tutorials/03_Blocking.html#devising-effective-blocking-rules-for-prediction) and [here](https://moj-analytical-services.github.io/splink/demos/tutorials/08_building_your_own_model.html#blocking-rules-for-prediction)

**Reply by mv802** [reply]
Thank you!
